cli/mender-artifact/dump.go

```diff
@@ -178,67 +179,69 @@ func dumpMetaData(metaDataDir string, dumpArgs *[]string, handlers map[int]handl
 	// this can recreate either type.
-	fmt.Printf("write module-image")
+	fmt.Fprintf(buf, "write module-image")
 
 	if ar.GetInfo().Version == 3 {
 		artProvs := ar.GetArtifactProvides()
-		fmt.Printf(" --artifact-name %s", artProvs.ArtifactName)
+		fmt.Fprintf(buf, " --artifact-name %s", artProvs.ArtifactName)
 		if len(artProvs.ArtifactGroup) > 0 {
-			fmt.Printf(" --provides-group %s", artProvs.ArtifactGroup)
+			fmt.Fprintf(buf, " --provides-group %s", artProvs.ArtifactGroup)
 		}
 
 		artDeps := ar.GetArtifactDepends()
 		if len(artDeps.ArtifactName) > 0 {
-			fmt.Printf(" --artifact-name-depends %s", strings.Join(artDeps.ArtifactName, " --artifact-name-depends "))
+			fmt.Fprintf(buf, " --artifact-name-depends %s", strings.Join(artDeps.ArtifactName, " --artifact-name-depends "))
 		}
-		fmt.Printf(" --device-type %s", strings.Join(artDeps.CompatibleDevices, " --device-type "))
+		fmt.Fprintf(buf, " --device-type %s", strings.Join(artDeps.CompatibleDevices, " --device-type "))
 		if len(artDeps.ArtifactGroup) > 0 {
-			fmt.Printf(" --depends-groups %s", strings.Join(artDeps.ArtifactGroup, " --depends-groups "))
+			fmt.Fprintf(buf, " --depends-groups %s", strings.Join(artDeps.ArtifactGroup, " --depends-groups "))
 		}
 
 	} else if ar.GetInfo().Version == 2 {
-		fmt.Printf(" --artifact-name %s", ar.GetArtifactName())
-		fmt.Printf(" --device-type %s", strings.Join(ar.GetCompatibleDevices(), " --device-type "))
+		fmt.Fprintf(buf, " --artifact-name %s", ar.GetArtifactName())
+		fmt.Fprintf(buf, " --device-type %s", strings.Join(ar.GetCompatibleDevices(), " --device-type "))
 	}
 
 	handlers := ar.GetHandlers()
 	handler := handlers[0]
 
-	fmt.Printf(" --type %s", handler.GetUpdateType())
+	fmt.Fprintf(buf, " --type %s", handler.GetUpdateType())
 
 	// Always add this flag, since we will write custom flags.
-	fmt.Printf(" --%s", noDefaultSoftwareVersionFlag)
+	fmt.Fprintf(buf, " --%s", noDefaultSoftwareVersionFlag)
 
 	provs := handler.GetUpdateOriginalProvides()
 	if provs != nil {
 		for key, value := range provs {
-			fmt.Printf(" --provides %s:%s", key, value)
+			fmt.Fprintf(buf, " --provides %s:%s", key, value)
 		}
 	}
 
 	deps := handler.GetUpdateOriginalDepends()
 	if deps != nil {
 		for key, value := range deps {
-			fmt.Printf(" --depends %s:%s", key, value)
+			fmt.Fprintf(buf, " --depends %s:%s", key, value)
 		}
 	}
 
 	// Always add this flag, since we will write custom flags.
-	fmt.Printf(" --%s", noDefaultClearsProvidesFlag)
+	fmt.Fprintf(buf, " --%s", noDefaultClearsProvidesFlag)
 
 	caps := handler.GetUpdateOriginalClearsProvides()
 	if caps != nil {
 		for _, value := range caps {
-			fmt.Printf(" --%s '%s'", clearsProvidesFlag, value)
+			fmt.Fprintf(buf, " --%s '%s'", clearsProvidesFlag, value)
 		}
 	}
 
 	if len(args) > 0 {
-		fmt.Printf(" %s\n", strings.Join(args, " "))
+		fmt.Fprintf(buf, " %s\n", strings.Join(args, " "))
 	}
+	fmt.Printf(strings.ReplaceAll(buf.String(), " ", separator))
```

Comment:
This is not future proof, it will replace all spaces, including ones that are intended in a single argument. Granted, it's not what we're trying to tackle here, but I think it's better to get this right from the start. The change should be simple though, just print the separator at the start of every argument instead of the space.